### PR TITLE
Chore: Add props for metric and error callbacks

### PR DESCRIPTION
**Pull request by JustinHoldstock**
@priyajeet Does this look like correct changes to accommodate handling preview errors and metrics? Do I need to make changes to the ES6 Wrapper, and if so, should those just be class properties that are passed, similar to `onInteraction()` but set as `noop` for the time being?

TODO: Tests!

### CODE REVIEW

**Comment by wenboyu2** [comment]
# Webpack Bundle Size Github Bot

| Bundle        | Diff | Before | After |
| ------------- |:----:|:------:|:-----:|
| Total | 0 B | 0 B | 0 B |

Compared against previously merged PR #255

**Powered by [webpack-bundle-size-github-bot](https://github.com/wenboyu2/webpack-bundle-size-github-bot/)**

**Comment by priyajeet** [comment]
This change should suffice, but if we eventually want someone to use the preview.js from UI Elements as opposed to preview.js from box-content-preview, then you will have to add it to the wrapper. So that the events are forwarded. See similar functions inside box-ui-elements/src/wrappers/ContentPicker.js

But if onError and onMetric are box use-case only, like only for webapp, then leaving them out of the wrapper is fine.

File: src/components/ContentPreview/ContentPreview.js [hunk 1]
```diff
@@ -66,7 +66,9 @@ type Props = {
     logoUrl?: string,
     sharedLink?: string,
     sharedLinkPassword?: string,
+    onError: Function,
```
Comment (priyajeet) [inline src/components/ContentPreview/ContentPreview.js]: these should be optional, no?

File: src/components/ContentPreview/ContentPreview.js [hunk 2]
```diff
@@ -87,6 +89,8 @@ class ContentPreview extends PureComponent<Props, State> {
     previewContainer: ?HTMLDivElement;
     mouseMoveTimeoutID: TimeoutID;
     rootElement: HTMLElement;
+    onError: Function;
```
Comment (priyajeet) [inline src/components/ContentPreview/ContentPreview.js]: Get rid of these

File: src/components/ContentPreview/ContentPreview.js [hunk 3]
```diff
@@ -140,6 +148,8 @@ class ContentPreview extends PureComponent<Props, State> {
             requestInterceptor,
             responseInterceptor
         });
+        this.onError = onError;
```
Comment (priyajeet) [inline src/components/ContentPreview/ContentPreview.js]: Remove these, use onError/onMetric directly from props

**Comment by boxcla** [comment]
# Webpack Bundle Size Github Bot

| Bundle        | Diff | Percentage Diff | Before | After |
| ------------- |:----:|:---------------:|:------:|:-----:|
| sidebar.js | 0 B| 0.00% | 486 KB | 486 KB |
| sidebar.css | 0 B| 0.00% | 24.6 KB | 24.6 KB |
| tree.js | 0 B| 0.00% | 498 KB | 498 KB |
| tree.css | 0 B| 0.00% | 10.6 KB | 10.6 KB |
| explorer.js | 1 KB| 0.15% | 682 KB | 683 KB |
| explorer.css | 0 B| 0.00% | 38.8 KB | 38.8 KB |
| picker.js | 0 B| 0.00% | 599 KB | 599 KB |
| picker.css | 0 B| 0.00% | 28.4 KB | 28.4 KB |
| uploader.js | 0 B| 0.00% | 501 KB | 501 KB |
| uploader.css | 0 B| 0.00% | 18.7 KB | 18.7 KB |
| preview.js | 0 B| 0.00% | 547 KB | 547 KB |
| preview.css | 0 B| 0.00% | 25.8 KB | 25.8 KB |
| Total | 1 KB | 0.03% | 3.38 MB | 3.38 MB |

Compared against previously merged PR #255

**Powered by [webpack-bundle-size-github-bot](https://github.com/wenboyu2/webpack-bundle-size-github-bot/)**

File: src/wrappers/ContentPreview.js [hunk 4]
```diff
@@ -7,13 +7,13 @@
 import React from 'react';
 import { render } from 'react-dom';
 import ES6Wrapper from './ES6Wrapper';
-import ContentPreviewComponent from '../components/ContentPreview/ContentPreview';
+import ContentPreviewResponsive from '../components/ContentPreview/ContentPreview';
```
Comment (DanDeMicco) [inline src/wrappers/ContentPreview.js]: Can this just be contentPreview?

**Pull request merged**